### PR TITLE
Add `visible_to` filter to the add UI for program actions/flows

### PR DESCRIPTION
**Pull request by nmajor25**
Shortcut Story ID: [sc-51292]

### CODE REVIEW

File: test/integration/patients/patient/dashboard.js
Comment (nmajor25) [inline test/integration/patients/patient/dashboard.js:1]: Goal is to cover these scenarios in the tests for both addable flows and actions:

1. `relationship.visible_to` does not exist (treated the same as being an empty array). Action/flow should be addable.
2. `relationship.visible_to` exists and contains the current user's team. Action/flow should be addable.
3. `relationship.visible_to` exists but does not contain the current user's team. Action/flow should not be addable.

File: src/js/entities-service/entities/program-actions.js [hunk 1]
```diff
@@ -129,15 +129,25 @@ const Collection = BaseCollection.extend({
       data: JSON.stringify({ data }),
     });
   },
+  isVisibleToCurrentUser(action) {
+    const visibleToList = action.get('_visible_to');
+    const currentUser = Radio.request('bootstrap', 'currentUser');
+    const currentUserTeamId = currentUser.get('_team');
+
+    if (!size(visibleToList)) return true;
```
Comment (nmajor25) [inline src/js/entities-service/entities/program-actions.js]: If `relationships.visible_to` is `null`/`undefined` or an empty `[]`, assume it's addable by every team/user.

File: test/integration/patients/patient/dashboard.js [hunk 2]
```diff
@@ -552,6 +568,18 @@ context('patient dashboard page', function() {
...
+      getFlow({
+        attributes: {
+          name: 'Should not show - not visible to current user team',
+          behavior: 'standard',
+          published_at: testTs(),
+          archived_at: null,
+        },
+        relationships: {
+          program: getRelationship(testProgramIds[1], 'programs'),
+          visible_to: { data: [{ type: 'teams', id: teamNurse.id }] },
```
Comment (nmajor25) [inline test/integration/patients/patient/dashboard.js]: Better way to do this?

File: src/js/entities-service/entities/program-actions.js [hunk 3]
```diff
@@ -129,15 +129,25 @@ const Collection = BaseCollection.extend({
       data: JSON.stringify({ data }),
     });
   },
+  isVisibleToCurrentUser(action) {
```
Comment (nmajor25) [inline src/js/entities-service/entities/program-actions.js]: Open to any better naming conventions for these functions/variables.

File: test/integration/patients/patient/dashboard.js (hunk 2 above)
Comment (shadowhand) [reply test/integration/patients/patient/dashboard.js]: If `teamNurse` has `id, type` defined, then we could do this ... ?

```suggestion
          visible_to: { data: [ teamNurse ] },
```

Comment (nmajor25) [reply test/integration/patients/patient/dashboard.js]: Yep that works 👍 . Via `visible_to: getRelationship([teamNurse])`. I forgot we had examples of doing that elsewhere in our tests.

File: src/js/entities-service/entities/program-actions.js [hunk 4]
```diff
@@ -129,15 +129,25 @@ const Collection = BaseCollection.extend({
...
+    const visibleToList = action.get('_visible_to');
+    const currentUser = Radio.request('bootstrap', 'currentUser');
+    const currentUserTeamId = currentUser.get('_team');
```
Comment (paulfalgout) [inline src/js/entities-service/entities/program-actions.js]: I think we should us `getTeam()` here.  we can use `.id`

File: src/js/entities-service/entities/program-actions.js (hunk 3 above)
Comment (paulfalgout) [reply src/js/entities-service/entities/program-actions.js]: I think we can move this function up into the action/flow such that it ends up `const isVisible = action.isVisibleToCurrentUser();`

File: test/integration/patients/patient/patient-flow.js [hunk 5]
```diff
@@ -424,6 +424,12 @@ context('patient flow page', function() {
   });
 
   specify('add action', function() {
+    const currentClinician = getCurrentClinician({
```
Comment (paulfalgout) [inline test/integration/patients/patient/patient-flow.js]: Can we flip these tests here and in patient-action to use the default `teamNurse` that the test `currentClinician` is by default so we don't need to set it?

**Review by paulfalgout — CHANGES_REQUESTED** [review]

File: test/integration/patients/patient/patient-flow.js (hunk 5 above)
Comment (nmajor25) [reply test/integration/patients/patient/patient-flow.js]: Good call, updated 👍

File: src/js/entities-service/entities/program-actions.js (hunk 4 above)
Comment (nmajor25) [reply src/js/entities-service/entities/program-actions.js]: Updated 👍

File: src/js/entities-service/entities/program-actions.js (hunk 3 above)
Comment (nmajor25) [reply src/js/entities-service/entities/program-actions.js]: Updated 👍

**Review by paulfalgout — APPROVED** [review]

**Pull request merged**